frontend-react/src/components/DataDashboard/DataDashboardTable/DataDashboardTable.tsx

```diff
@@ -1,103 +1,113 @@
 
 import { FeatureName } from "../../../AppRouter";
 import { EventName, trackAppInsightEvent } from "../../../utils/Analytics";
-import usePagination from "../../../hooks/UsePagination";
 import { RSReceiver } from "../../../config/endpoints/settings";
 import { useOrganizationReceiversFeed } from "../../../hooks/UseOrganizationReceiversFeed";
 import Spinner from "../../Spinner";
 import { NoServicesBanner } from "../../alerts/NoServicesAlert";
-import { FilterManager } from "../../../hooks/filters/UseFilterManager";
-import { PaginationProps } from "../../Table/Pagination";
-import Table, { ColumnConfig, TableConfig } from "../../Table/Table";
+import Pagination from "../../Table/Pagination";
 import TableFilters from "../../Table/TableFilters";
-import {
-    useOrgDeliveries,
-    DataDashboardAttr,
-} from "../../../hooks/network/DataDashboard/DataDashboardHooks";
-import { RSDelivery } from "../../../config/endpoints/dataDashboard";
 import ReceiverServices from "../ReceiverServices/ReceiverServices";
+import useReceiverDeliveries, {
+    DeliveriesAttr,
+} from "../../../hooks/network/DataDashboard/UseReceiverDeliveries";
+import AdminFetchAlert from "../../alerts/AdminFetchAlert";
+import { Table } from "../../../shared/Table/Table";
+import { getSlots } from "../../../hooks/UsePagination";
+import { PageSettingsActionType } from "../../../hooks/filters/UsePages";
+import { SortSettingsActionType } from "../../../hooks/filters/UseSortOrder";
 import { formatDateWithoutSeconds } from "../../../utils/DateTimeUtils";
+import { USLink } from "../../USLink";
 
-const extractCursor = (d: RSDelivery) => d.batchReadyAt;
+function DashboardFilterAndTable({
+    receiverServices,
+    activeService,
+    setActiveService,
+}: {
+    receiverServices: RSReceiver[];
+    activeService: RSReceiver;
+    setActiveService: Dispatch<SetStateAction<RSReceiver | undefined>>;
+}) {
+    const featureEvent = `${FeatureName.DATA_DASHBOARD} | ${EventName.TABLE_FILTER}`;
 
-interface DashboardTableContentProps {
-    receivers: RSReceiver[];
-    activeService: RSReceiver | undefined;
-    handleSetActive: (v: string) => void;
-    filterManager: FilterManager;
-    paginationProps?: PaginationProps;
-    isLoading: boolean;
-    receiverList: RSDelivery[] | undefined;
-}
+    const handleSetActive = (name: string) => {
+        setActiveService(receiverServices.find((item) => item.name === name));
+    };
 
-const DashboardTableContent: React.FC<DashboardTableContentProps> = ({
-    receivers,
-    activeService,
-    handleSetActive,
-    filterManager,
-    paginationProps,
-    isLoading,
-    receiverList,
-}) => {
-    if (isLoading) return <Spinner />;
+    const {
+        data: results,
+        filterManager,
+        isLoading,
+    } = useReceiverDeliveries(activeService.name);
 
-    const featureEvent = `${FeatureName.DAILY_DATA} | ${EventName.TABLE_FILTER}`;
+    if (isLoading || !results) return <Spinner />;
```

Comment:
do we want to return a Spinner or a error message for no results? Maybe an opportunity to log?

Reply:
This spinner will only appear when fetching the table data. If there are no results then the table component displays a message.